docs/type-long-integer.md

```diff
@@ -0,0 +1,96 @@
+# Long Integer Type
+
+## Summary
+
+A builtin type to represent 64-bit integers.
+
+## Motivation
+
+Current methods of representing 64-bit integers in Luau are done with high-low pairs or by splitting them among vectors which leads to poor ergonomics and may not be sufficient for cases where performance is important.
+
+In the case of a high-low pair implementation each individual number takes 16 bytes (or 24) and needs to be be handled together which can be confusing.
+
+In the case of vector implementations, ergonomics are improved by storing the entire integer in one value however you have the same issues of implementation complexity and lack of integration with the type system.
+
+In both cases, performance is lacking compared to what could be provided by a native implementation.
+
+As Luau grows the restriction to doubles will be an increasing pain point for any use case which involves a 64-bit integer.
+System support for 64-bit integers is ubiquitous their use is widespread.
+
+While a runtime could implement 64-bit integers with userdata, performing an allocation for each integer (which can easily fit within an existing Luau value) is an onerous requirement.
```

Comment:
It is not an onerous requirement in a language like Luau.
Userdata path is a completely normal way to implement custom types for the embedder.
Languages similar to Luau also take the same boxed garbage-collectable object paths in their implementation and Luau has an efficient allocator to support this.